homework-submission/abdullahReact/main.js

```diff
@@ -0,0 +1,98 @@
+const toDoItems = [
+  {
+    id: 1,
+    description: 'Get out of bed',
+    deadline: '2017-09-11',
+    done: true,
+  },
+  {
+    id: 2,
+    description: 'Brush teeth',
+    deadline: '2017-09-10',
+    done: false,
+  },
+  {
+    id: 3,
+    description: 'Eat breakfast',
+    deadline: '2017-09-09',
+    done: false,
+  },
+];
+
+//the goal of this component is to define/create a reusable list item that uses own function to set state.
+// in this case to set as done or not done by clicking
```

Comment:
Could you describe all the props this component can receive and what their goal is / what the effect is of the prop? This is very useful for other developers using your component.

Reply:
thanks for advise. I added more description

Reply:
Great, totally fine like this. Just be aware that saying "// props.done is to initialize state.done." doesn't say a lot about the effect is of the prop. In my opinion a better comment would be: "// props.done gives the ListItem a line-through when it's true"

Reply:
yes it makes sense. I corrected that line